controllers/services/monitoring/monitoring_controller_actions.go

```diff
@@ -5,44 +5,109 @@ import (
 	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 )
 
+// initialize handles all pre-deployment configurations.
 func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	log := logf.FromContext(ctx)
+	// Only handle manifests setup and initial configurations
+	platform := rr.Release.Name
+	switch platform {
+	case cluster.ManagedRhoai:
+		// Only set prometheus configmap path
+		rr.Manifests = []odhtypes.ManifestInfo{
+			{
+				Path:       odhdeploy.DefaultManifestPath,
+				ContextDir: "monitoring/prometheus/apps",
+			},
+		}
+
+	default:
+		log.Info("Monitoring enabled, won't apply changes in this mode", "cluster", platform)
+	}
+
 	return nil
 }
 
-func updateStatus(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
-	d, ok := rr.Instance.(*serviceApi.Monitoring)
-	if !ok {
-		return errors.New("instance is not of type *services.Monitoring")
+func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// Map component names to their rule prefixes
+	dscList := &dscv1.DataScienceClusterList{}
+	if err := rr.Client.List(ctx, dscList); err != nil {
+		return fmt.Errorf("failed to list DSC: %w", err)
+	}
+	if len(dscList.Items) == 0 {
+		return nil
+	}
+	dsc := &dscList.Items[0]
+	componentRules := map[string]string{
+		componentApi.DashboardComponentName:            "rhods-dashboard",
+		componentApi.WorkbenchesComponentName:          "workbenches",
+		componentApi.KueueComponentName:                "kueue",
+		componentApi.CodeFlareComponentName:            "codeflare",
+		componentApi.DataSciencePipelinesComponentName: "data-science-pipelines-operator",
+		componentApi.ModelMeshServingComponentName:     "model-mesh",
+		componentApi.RayComponentName:                  "ray",
+		componentApi.TrustyAIComponentName:             "trustyai",
+		componentApi.KserveComponentName:               "kserve",
+		componentApi.TrainingOperatorComponentName:     "trainingoperator",
+		componentApi.ModelRegistryComponentName:        "model-registry-operator",
+		componentApi.ModelControllerComponentName:      "odh-model-controller",
 	}
 
-	// url
-	rl := routev1.RouteList{}
-	err := rr.Client.List(
-		ctx,
-		&rl,
-		client.InNamespace(rr.DSCI.Spec.Monitoring.Namespace),
-		client.MatchingLabels(map[string]string{
-			labels.PlatformPartOf: serviceApi.MonitoringServiceName,
-		}),
-	)
+	err := cr.ForEach(func(ch cr.ComponentHandler) error {
+		var enabled bool
+		ci := ch.NewCRObject(dsc)
+		// read the component instance to get tha actual status
+		err := rr.Client.Get(ctx, client.ObjectKeyFromObject(ci), ci)
+		switch {
+		case k8serr.IsNotFound(err):
+			enabled = false
+		case err != nil:
+			enabled = false
```

Comment:
nit: this is ineffective since it returns just after

Reply:
updated